### DISCUSSION TITLE
BloomFilter capacity limitation

### DISCUSSION THREAD

**Discussion by crazylazyowl** [apache/kvrocks#2541] [Q&A] [answered]
Good day!

I have a question about the BloomFilter support.

I tried migrating my project from Redis to KVRocks and encountered an issue where it’s not possible to create a BloomFilter with a capacity greater than 2^32. I checked the code [[click](https://github.com/apache/kvrocks/blob/399961dd847faf0512b5cddb24bdce2977b07796/src/commands/cmd_bloom_filter.cc#L53)] and noticed that when parsing the `BF.RESERVE` command, you are using `ParseInt<uint32_t>` for the capacity argument.

As a result:
```
127.0.0.1:6666> bf.reserve x 0.001 20000000000
(error) ERR Bad capacity
127.0.0.1:6666> bf.reserve x 0.001 10000000000
(error) ERR Bad capacity
127.0.0.1:6666> bf.reserve x 0.001 9000000000
(error) ERR Bad capacity
127.0.0.1:6666> bf.reserve x 0.001 8000000000
(error) ERR Bad capacity
127.0.0.1:6666> bf.reserve x 0.001 7000000000
(error) ERR Bad capacity
127.0.0.1:6666> bf.reserve x 0.001 6000000000
(error) ERR Bad capacity
127.0.0.1:6666> bf.reserve x 0.001 5000000000
(error) ERR Bad capacity
127.0.0.1:6666> bf.reserve x 0.001 4000000000
OK
(0.65s)
```

Is this a deliberate limitation? Or we can consider such behaviour as a bug?

Thank you!

**Comment by PragmaTwice** [comment]
As you can see, the `parse_capacity` is typed uint32_t, which means it cannot be larger than 2^32.

It's expected.

**Reply by crazylazyowl** [reply]
So, it's not a bug and you won't change it to 2**64?

**Reply by PragmaTwice** [reply, marked as answer]
If you check https://kvrocks.apache.org/community/data-structure-on-rocksdb#bloom-filter, you can get more idea about how it's stored on disk.

I'm not the guy who design this part, but from the design you can see it's 32bit based. If you think that it's not enough, maybe you can provide your use case so that we can have a discussion : )

**Reply by crazylazyowl** [reply]
Thank you :) I'll check it
